src/main/java/org/eolang/compiler/Program.java

```diff
@@ -53,11 +56,11 @@
     }
 
     /**
-     * Compile it to Java.
-     * @return Java code
+     * Compile it to Java and save to the directory.
+     * @param dir Directory to save to
      * @throws IOException If fails
      */
-    public String java() throws IOException {
+    public void save(final Path dir) throws IOException {
```

Comment:
there are two of these with different names. Both seem like they should defer to their own implementation class; the names should be distinct as both serve a separate purpose.